src/Sulu/Bundle/SnippetBundle/Controller/SnippetController.php

```diff
@@ -274,7 +273,7 @@ public function putAction(Request $request, $id)
      */
     public function deleteAction(Request $request, $id)
     {
-        $locale = $this->getLocale($request);
+        $locale = $this->getLocale($request) ?? 'de_de';
```

Comment:
we should also not default here to anything hardcoded

Reply:
Well the getLocale function shouldn't return null or else it will cause other problemes. But for now we'll just need to expand the baseline for those errors. 'Cause we can't have function to return something nullable but the places where the value is used not accept null. Would that work for you?

Reply:
Need to check the Permission Request Listener. If we could fallback to `$request->getLocale` in `$this->getLocale()` or if we add a `bool $force` to `getLocale()` method for the cases where it is required.

Reply:
I don't like the adding a `$force` parameter idea, since you basically completely changes the function.
 
What should happen when the locale is empty? Throw an exception?

Reply:
> I don't like the adding a $force parameter idea, since you basically completely changes the function.

The `$force = false` is required as if not forcing the `getLocale` is required to return `null` for the SecurityListener. With the  `bool $force = false` we make sure the method behave the same as befor and only call with with `, true` where we 100% require a locale.

The exception should be this: https://github.com/sulu/sulu/blob/08a40b7de9c1b3f9a6c42edc928ee8e4ff9ea97c/src/Sulu/Component/Rest/RequestParametersTrait.php#L38

Reply:
Makes sense. How about we make a new function for that and deprecate the old one. So that we can have a clean cut between the old and the new behaviour.

Reply:
I need to think about that. At current state I would concentrate in this PR on the PHPDoc and keep the other code as it is so we have code changes itself then in own pull request make searching for issues later easier.